src/main/java/org/craftercms/studio/impl/v2/service/notification/NotificationServiceImpl.java

```diff
@@ -104,49 +103,106 @@ public void init() {
+            templateModel.put(TEMPLATE_MODEL_REVIEWER, reviewerUser);
+
+            // Keeping these for backwards compatibility
+            // Prefer reviewer
+            templateModel.put(TEMPLATE_MODEL_USER_THAT_REJECTS, reviewerUser);
+            // Prefer package.reviewerComment
+            templateModel.put(TEMPLATE_MODEL_REJECTION_REASON, publishPackage.getReviewerComment());
+
+            String email = submitterUser.get(KEY_EMAIL).toString();
+            notify(siteId, List.of(email), NOTIFICATION_KEY_CONTENT_REJECTED, templateModel);
+        } catch (UserNotFoundException e) {
+            logger.error("Failed to send content rejection notification because user was not found", e);
+        } catch (Exception e) {
+            logger.error("Failed to send content rejection notification for site '{}'", siteId, e);
+        }
     }
 
     @Override
-    @Valid
-    public void notifyContentApproval(@ValidateStringParam final String site,
-                                      @ValidateStringParam final String submitter,
-                                      final List<String> itemsSubmitted,
-                                      @ValidateStringParam final String approver,
-                                      final ZonedDateTime scheduleDate) {
+    public void notifyPackageSubmission(final PublishPackage publishPackage, final Collection<String> paths) {
+        String siteId = publishPackage.getSite().getSiteId();
+        logger.debug("Sending content submission notification for site '{}', package '{}'", siteId, publishPackage.getId());
         try {
-            final Map<String, Object> submitterUser = securityService.getUserProfile(submitter);
+            final NotificationConfigTO notificationConfig = getNotificationConfig(siteId);
+            final Map<String, Object> submitterUser = securityService.getUserProfile(publishPackage.getSubmitter().getUsername());
             Map<String, Object> templateModel = new HashMap<>();
-            templateModel.put(TEMPLATE_MODEL_FILES, convertPathsToContent(site, itemsSubmitted));
-            templateModel.put(TEMPLATE_MODEL_SUBMITTER_USER, submitter);
-            templateModel.put(TEMPLATE_MODEL_APPROVER, securityService.getUserProfile(approver));
-            templateModel.put(TEMPLATE_MODEL_SCHEDULED_DATE, (scheduleDate == null) ?
-                    null : Date.from(scheduleDate.toInstant()));
-            notify(site, singletonList(submitterUser.get(KEY_EMAIL).toString()), NOTIFICATION_KEY_CONTENT_APPROVED,
+            templateModel.put(TEMPLATE_MODEL_PACKAGE, publishPackage);
+            templateModel.put(TEMPLATE_MODEL_FILES, convertPathsToContent(siteId, paths));
+            templateModel.put(TEMPLATE_MODEL_SUBMITTER, submitterUser);
+            templateModel.put(TEMPLATE_MODEL_SUBMISSION_COMMENTS, publishPackage.getSubmitterComment());
+
+            // Keeping these for backwards compatibility
```

Comment:
Do we need a ticket to update all blueprints to use the new notification variables?

Reply:
https://github.com/craftercms/craftercms/issues/7557